### DISCUSSION TITLE
sea_orm_cli: Enums could implement Copy

### DISCUSSION THREAD

**Discussion by junbl** [SeaQL/sea-orm#668] [General]
## Summary

The enums generated by `sea_orm_cli` (those placed in `sea_orm_active_enums`, referencing databse enums) can derive copy, since they never have struct variants. 

## Motivation

The [language docs](https://doc.rust-lang.org/std/marker/trait.Copy.html#when-should-my-type-be-copy) have this to say:
>Generally speaking, if your type can implement Copy, it should. Keep in mind, though, that implementing Copy is part of the public API of your type. If the type might become non-Copy in the future, it could be prudent to omit the Copy implementation now, to avoid a breaking API change.

If the databases were to support more complex enum variants, then this may no longer be true. However, this seems unlikely.

## Additional Information

To be safe, it could be toggled with an optional flag (cf. `--with-serde`).

**Comment by billy1624** [comment]
Hey @e-rhodes, I agree with what you have suggested!

> The enums generated by `sea_orm_cli` (those placed in `sea_orm_active_enums`, referencing databse enums) can derive copy, since they never have struct variants.

Yes, this is true for ordinary user but we have to cater the need of advance users.

> To be safe, it could be toggled with an optional flag (cf. `--with-serde`).

I would say we can safely derive `Copy` for all generated Enums. For advance users, they simply have to update the Enums in `sea_orm_active_enums` file. And perhaps remove some of the `Copy` derive if they want to, it would be a easy "replace all" for them with the help of code editor.

Sounds good?

CC @tyt2y3

**Comment by tyt2y3** [comment]
> If the type might become non-Copy in the future, it could be prudent to omit the Copy implementation now, to avoid a breaking API change.

Yes, I am still considering whether I'd extend the API by using tuple variants in enums
